src/App.tsx

```diff
@@ -15,36 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type FilterQuery = {
+  sortType: SortType | '';
+  isReversed: boolean;
+};
+
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+function getPreperedArray(
+  goods: string[],
+  { sortType, isReversed }: FilterQuery,
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.Alphabet:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortType(SortType.Default);
+  };
+
+  const visibleGoods = getPreperedArray(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortType(SortType.Alphabet)}
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortType !== SortType.Alphabet,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortType(SortType.Length)}
```

Comment:
suggest to make handler factory in component and use for both sort buttons.

```
const makeHandleSort = (sortType: SortType) => () => setSortType(sortType)

```
```suggestion
          onClick={makeHandleSort(SortType.Length)}
```